nntrainer/graph/network_graph.cpp

```diff
@@ -468,11 +468,16 @@ void NetworkGraph::setOutputLayers() {
         continue;
       for (unsigned int j = 0; j < layer_i->input_layers.size(); ++j) {
         if (istrequal(layer_i->input_layers[j], layer_idx->getName())) {
+          bool already_exist = false;
           for (unsigned int k = 0; k < layer_idx->output_layers.size(); ++k) {
-            if (!istrequal(layer_idx->output_layers[k], layer_i->getName()))
-              continue;
+            if (!istrequal(layer_idx->output_layers[k], layer_i->getName())) {
+              already_exist = true;
+              break;
+            }
```

Comment:
slightly off the topic though, I wonder what this loop is trying to to. It look essentially noop to me.

Reply:
it is trying to avoid duplicates in `layer_idx->output_layers`.

Reply:
Ah, I see, nice catch 👍